pyamg/gallery/fem.py

```diff
@@ -867,25 +867,28 @@ def divform(mesh):
     return BX, BY
 
 
-def applybc(A, b, mesh, bc):
+def applybc(A, b, mesh, bc, remove_dirichlet=False):
     """Apply boundary conditions.
 
     Parameters
     ----------
     A : sparse matrix
-        Fully assembled sparse matrix
+        Fully assembled sparse matrix.
     b : ndarray
-        Fully assembled right-hand side
+        Fully assembled right-hand side.
     mesh : Mesh
-        Mesh object
+        Mesh object.
     bc : list
-       list of boundary conditions
-       bc = [bc1, bc2, ..., bck]
-       where bck = {'id': id,    a list of vertices for boundary "k"
-                     'g': g,     g = g(x,y) is a function for the vertices on boundary "k"
-                   'var': var    the variable, given as a start in the dof list
-                'degree': degree degree of the variable, either 1 or 2
-                   }
+       List of boundary conditions: bc = [bc1, bc2, ..., bck],
+       where
+
+           bck = {'id': id,    # a list of vertices for boundary "k"
+                   'g': g,     # g = g(x,y) is a function for the vertices on boundary "k"
+                 'var': var    # the variable, given as a start in the dof list
```

Comment:
I found it a little hard to see exactly what I need to set "var" and "degree" to.  Maybe give an example of what to set these for all dirichlet, and half dirichlet / half neumann?

Reply:
added a partial example.  but expanding/adding more, I've added an issue #441